src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Filters/LiquidViewFilters.cs

```diff
@@ -51,11 +59,11 @@ public static async ValueTask<FluidValue> NewShape(FluidValue input, FilterArgum
         {
             if (!context.AmbientValues.TryGetValue("ShapeFactory", out dynamic shapeFactory))
```

Comment:
I think it can be a `IShapeFactory` here. It's just that the default implementation has some dynamic helpers to create shapes by method invocation, but we don't use them here.

Reply:
Now trusting to be `IShapeFactory`